Update amazon-s3-connector-config-topics (#868)
mulesoft/docs-connectors: amazon-s3/5.6/modules/ROOT/pages/amazon-s3-connector-config-topics.adoc

[source,json,linenums]
----
{
  "Version": "2012-10-17",
  "Id": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME/SQSDefaultPolicy",
  "Statement": [
    {
      "Sid": "Sid1593770403887",
      "Effect": "Allow",
      "Principal": "*",
      "Action": "SQS:SendMessage",
      "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME",
      "Condition": {
        "ArnLike": {
          "aws:SourceArn": "arn:aws:s3:::BUCKET_NAME"
        }
      }
    }
  ]
}
----

[[queue-created-by-source]]
== Queue Created by Source Operation

If the queue is created by the source operation, the queue is created with the prefix `MULE-S3-TRIGGER`. 

In this scenario, the user must have permission to create a queue and to also set the queue attributes:

[source,json,linenums]
----
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "sqs:DeleteMessage",
                "sqs:ReceiveMessage",
                "sqs:GetQueueAttributes",
                "sqs:CreateQueue",

>                 "sqs:SetQueueAttributes"

Author: missing one action: please add there **"sqs:ReceiveMessage"**

> To encrypt objects that you want to store in S3 buckets using customer managed master keys, specify a Customer Master Key ID in the `KMS Master Key` field in the `Create Object` configuration.

Reviewer: ```suggestion
To encrypt objects that you want to store in S3 buckets using customer-managed master keys, specify a Customer Master Key ID in the `KMS Master Key` field in the `Create Object` configuration.
```

Reviewer: How do creating an object operation and master key--the main tasks of this topic set, according to the Next Steps text, specifically relate to the following subtopics?
It wasn't until I got to the final sentence that I realized that the two tasks here were
1. Configure master key
2. Provide credentials
Should those be explicitly stated using subheadings, and the topic arranged around this?

Reviewer: Please ignore edits below that reveal that I did not understand the focus of this topic when I first edited it; I was too lazy to remove them, and maybe some make sense.

> == Prerequisites for Source Operations

Reviewer: Should this be moved to the beginning of the queue created by source operations topic?
During my first reading, I thought that this topic set was arranged around
1. Queue exists
2. Source operation creates queue

There are two scenarios for the `On New Object` and `On Deleted Object` source operations:

> . The <<queue-is-created,queue is created>> and specified

Reviewer: Is "and specified" explained in subsequent text?

> . The <<queue-created-by-source,queue will be created>> by the source

Reviewer: ```suggestion
. The <<queue-created-by-source,queue is created>> by the source
```

[[queue-is-created]]

> == Queue is Already Created

Reviewer: ```suggestion
== Queue Is Already Created
```
It's not the length of the word so much as the type (except prepositions: 5 letters or more take an init cap).
Verbs and nouns are always capped.

> To correctly use a source operation that already exists, you must enable these actions for the Amazon SQS queue:

Author: Here we need to say that User must have those actions enabled. The word **user** is important here.

Reviewer: The title says the queue already exists, but the first sentence says that a source operation already exists: which is correct?

Author: My original sentence says: In order to properly use source operation, the user needs to have enabled these Actions for SQS. Source operation is something connector provides.
This sentence is not correct: To correctly use a source operation that already exists, you must enable these actions for the Amazon SQS queue:

> To correctly use a source operation that already exists, you must enable these actions for the Amazon SQS queue:

Reviewer: ```suggestion
To correctly use a source operation that already exists, enable these actions for the Amazon SQS queue:
```

[source,json,linenums]
----
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "sqs:DeleteMessage",
                "sqs:GetQueueUrl",
                "sqs:ReceiveMessage",
                "sqs:GetQueueAttributes"
            ],
            "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME"
        }
    ]
}
----

> Additionally, to enable the queue to receive messages when a new object is created in an S3 bucket, you must create permission for the action in this format:

Author: you must create permission for the action in this format -> you must create policy for the queue in this format

> Additionally, to enable the queue to receive messages when a new object is created in an S3 bucket, you must create permission for the action in this format:

Reviewer: ```suggestion
Additionally, to enable the queue to receive messages when a new object is created in an S3 bucket, create a policy for the queue in this format:
```

[source,json,linenums]
----
{
  "Version": "2012-10-17",
  "Id": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME/SQSDefaultPolicy",
  "Statement": [
    {
      "Sid": "Sid1593770403887",
      "Effect": "Allow",
      "Principal": "*",
      "Action": "SQS:SendMessage",
      "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:QUEUE_NAME",
      "Condition": {
        "ArnLike": {
          "aws:SourceArn": "arn:aws:s3:::BUCKET_NAME"
        }
      }
    }
  ]
}
----

[[queue-created-by-source]]

> == Queue Created by Source Operation

Reviewer: ```suggestion
== Queue Is Created by Source
```
parallelism, agreement with intro list

> If the queue is created by the source operation, the queue is created with the prefix `MULE-S3-TRIGGER`. 

Reviewer: ```suggestion
If the queue is created by the source operation, the queue name has the prefix `MULE-S3-TRIGGER`. 
```

Reviewer: But why is this the first thing said in this topic?

Author: Because when the source operation is invoke (and the queue is not already created on aws console) the first thing is that it will create queue with this prefix.

> If the queue is created by the source operation, the queue is created with the prefix `MULE-S3-TRIGGER`. 

Reviewer: Do we need to say this if it is part of the syntax example?
We could also say "Queues created using a source operation require the prefix..." and tell WHY?

> In this scenario, you must have permission to create a queue and to also set the queue attributes:

Author: the **user** must have a permission

Reviewer: If the queue is created by the source, why do YOU have to do anything?

Maybe we need to qualify this to "the queue is created by using a source operation"?

Are the two scenarios then...:

The queue already exists
The user creates the queue

Reviewer: ```suggestion
Creating the queue by using a source operation requires permission to both create the queue and set the queue attributes:
```

Author: Because Aws works that way. There is difference between user permission and queue policy. When admin on aws creates a user by default he does not have any permission at all. So we need to give him some kind of permission. In this case we need to give him permissions (the minimum) to create queue and other actions in order to invoke operations. 

example: operation provided by SQS - create queue - user needs permission with action **create queue**
source operation provided by S3 - behind the scene we are using create queue, delete message operations etc. Therefore, user who is using connector does not know what type of permission he needs for source operation

[source,json,linenums]
----
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "VisualEditor0",
            "Effect": "Allow",
            "Action": [
                "sqs:DeleteMessage",
                "sqs:ReceiveMessage",
                "sqs:GetQueueAttributes",
                "sqs:CreateQueue",
                "sqs:SetQueueAttributes"
            ],
            "Resource": "arn:aws:sqs:REGION:ACCOUNT_ID:MULE-S3-TRIGGER-*"
        }
    ]
}
----

> When the queue is created by the source operation, an S3 policy is added to the queue when the S3 object is created in the bucket, in the following format:

Author: S3 policy -> SQS policy

Reviewer: This tech review comment confused me: are we documenting S3 or SQS policies, or both, and do we need to clarify that earlier in this topic set?

Is the policy added automatically?

Is this what we mean:
```suggestion
Queues created by using a source operation automatically receive an SQS policy when the SQS object is created in the bucket, in the following format:
```

Reviewer: But...what about the S3 policy referred to in the following section?
I am confused: we show one policy format here, and another below, and it is unclear who creates the policy or creates objects in the bucket. But if this is just my lack of product knowledge, well, you know what to do....  ;-)

Author: We are documenting S3 source operations but behind the scene these source operations are using SQS to notify when a new object is created on S3 bucket. S3 policy user needs to add to his created bucket. and we have two scenarios.. 1. queue is already created on aws console therefore user does not need a permission (action) to create queue. 2. If the queue is not created on aws therefore source operation behind the scenes will create a queue and that's why user need a permission (action) to create queue. If the user does not have a permission to create queue in the console he will see access denied.

[source,json,linenums]
----
{
  "Version": "2012-10-17",
  "Id": "example-ID",
  "Statement": [
    {
      "Sid": "example-statement-ID",
      "Effect": "Allow",
      "Principal": {
        "AWS": "*"
    },
      "Action": "SQS:SendMessage",
      "Resource":QUEUE_ARN,
      "Condition": {
        "ArnLike": {
          "aws:SourceArn": "BUCKET_ARN"
        }
      }
    }
  ]
}
----

> == S3 Policy Bucket

Reviewer: S3 or SQS, per your SME?

Or are there two? If so, we should introduce two policies rather than referring to only one.

> For the S3 policy, we use this form for both cases:

Reviewer: "both cases"? Is this what we mean:
```suggestion
An S3 policy uses the following format, whether already created or created using a source operation:
```

[source,json,linenums]
----
{
    "Version": "2012-10-17",
    "Id": "Policy1593761427184",
    "Statement": [
        {
            "Sid": "Stmt1593760119344",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::ACCOUNT_ID:user/test"
            },
            "Action": "SPECIFIC_ACTION",
            "Resource": "arn:aws:s3:::BUCKET_NAME/*"
        },
        {
            "Sid": "Stmt1593760259223",
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::ACCOUNT_ID:user/test"
            },
            "Action": ["s3:PutBucketNotification", "s3:GetBucketNotification"],
            "Resource": "arn:aws:s3:::BUCKET_NAME"
        }
    ]
}
----

> In this example, the *SPECIFIC_ACTION* is `s3:PutObject` or `s3:DeleteObject`, which is based on the specified source operation.

Reviewer: ```suggestion
In this example, *SPECIFIC_ACTION* is `s3:PutObject` or `s3:DeleteObject`, depending on the specified source operation.
```

== Next Step

> After you understand how to configure a master key and provide credentials, you can try out the xref:amazon-s3-connector-examples.adoc[Example].

Reviewer: It wasn't until I got to this sentence that I realized that the two tasks here were
1. Configure master key
2. Provide credentials
Should those be explicitly stated using subheadings above, and the topic arranged around this?
